app/controllers/devise_controller.rb

```diff
@@ -167,7 +167,7 @@ def devise_i18n_options(options)
 
   # Get message for given
   def find_message(kind, options = {})
-    options[:scope] = "devise.#{controller_name}"
+    options[:scope] = "devise.#{controller_name}" unless options[:scope]
```

Comment:
@josevalim you are right that is more elegant:)

I am always kind of scare in using the monad style (there are different opinion of what really ||= means, but in this case there is no possibility of been wrong):

I am going to change the pull request and resubmit:)